source/Applications/Advanced/TransformViaArucoMarker/TransformViaArucoMarker.cpp

```diff
@@ -0,0 +1,308 @@
+/*
+This example shows how to transform a point cloud using the pose of an aruco marker.
+*/
+
+#include <Zivid/Experimental/Calibration.h>
+#include <Zivid/Visualization/Visualizer.h>
+#include <Zivid/Zivid.h>
+
+#include <algorithm>
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+
+#include <aruco_nano.h>
+#include <cmath>
+#include <iostream>
+
+namespace
+{
+    struct Line2d
+    {
+        double slope;
+        double intercept;
+    };
+
+    int clamp(const int n, const int lower, const int upper)
+    {
+        return std::max(lower, std::min(n, upper));
+    }
+
+    Line2d fitLine(const cv::Point2d &p1, const cv::Point2d &p2)
+    {
+        const auto slope{ (p2.y - p1.y) / (p2.x - p1.x) };
+        const auto intercept{ p1.y - slope * p1.x };
+        return { slope, intercept };
+    };
+
+    cv::Mat pointCloudToBGR(const Zivid::PointCloud &pointCloud)
+    {
+        auto rgb = cv::Mat(pointCloud.height(), pointCloud.width(), CV_8UC4); // NOLINT(hicpp-signed-bitwise)
+        pointCloud.copyData(
+            reinterpret_cast<Zivid::ColorRGBA *>(rgb.data)); // NOLINT(cppcoreguidelines-pro-type-reinterpret-cast)
+        auto bgr = cv::Mat(pointCloud.height(), pointCloud.width(), CV_8UC4); // NOLINT(hicpp-signed-bitwise)
+        cv::cvtColor(rgb, bgr, cv::COLOR_RGBA2BGR);
+
+        return bgr;
+    }
+
+    std::vector<cv::Point2f> estimate2DMarkerCenters(const aruconano::Marker &markers)
+    {
+        if(markers.empty())
+        {
+            std::cout << "No markers detected";
+            return {};
+        }
+
+        std::vector<cv::Point2f> markerCenters2D;
+        markerCenters2D.reserve(markers.size());
+
+        auto p11 = markers[0];
+        auto p12 = markers[1];
+        auto p21 = markers[3];
+        auto p22 = markers[2];
+
+        const auto backDiagonal{ fitLine(p11, p22) };
+        const auto forwardDiagonal{ fitLine(p21, p12) };
```

Comment:
What made you use these variable names? They are not intuitive to me.

Reply:
Added a comment explaining this step. But i'm not sure about the best naming for these two...